src/components/picker/types.tsx

```diff
@@ -48,85 +48,68 @@ export interface PickerSearchStyle {
   selectionColor?: string;
 }
 
-export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> & {
-  /* ...TextField.propTypes, */
-  /**
-   * Use dialog instead of modal picker
-   */
-  useDialog?: boolean;
+export type PickerPropsDeprecation = {
   /**
    * @deprecated
    * Temporary prop required for migration to Picker's new API
    */
   migrate?: boolean;
   /**
-   * Pass for different field type UI (form, filter or settings)
-   */
-  fieldType?: PickerFieldTypes | `${PickerFieldTypes}`;
-  /**
-   * Picker current value in the shape of {value: ..., label: ...}, for custom shape use 'getItemValue' prop
+   * @deprecated
+   * A function that extract the unique value out of the value prop in case value has a custom structure (e.g. {myValue, myLabel})
    */
-  value?: PickerValue;
+  getItemValue?: (value: PickerValue) => any;
   /**
-   * Callback for when picker value change
+   * @deprecated
+   * A function that extract the label out of the value prop in case value has a custom structure (e.g. {myValue, myLabel})
    */
-  onChange?: (value: PickerValue) => void;
+  getItemLabel?: (value: PickerValue) => string;
   /**
-   * SINGLE mode or MULTI mode
+   * @deprecated
+   * Callback for modal onShow event
+   * Instead pass onShow via customPickerProps.modalProp | pickerModalProps
    */
-  mode?: PickerModes | `${PickerModes}`;
+  onShow?: () => void;
   /**
-   * Limit the number of selected items
+   * @deprecated
+   * instead pass items
    */
-  selectionLimit?: number;
+  children?: ReactNode | undefined;
   /**
-   * Adds blur effect to picker modal (iOS only)
+   * @deprecated
+   * Render a custom header for Picker's dialog
+   * instead use renderCustomOverlayHeader
    */
-  enableModalBlur?: boolean;
+  renderCustomDialogHeader?: (callbacks: {onDone?: () => void; onCancel?: () => void}) => React.ReactElement;
   /**
+   * @deprecated
    * Render custom picker - input will be value (see above)
    * Example:
    * renderPicker = (selectedItem) => {...}
+   * instead use renderInput
    */
   renderPicker?: RenderPicker;
   /**
-   * Render custom picker item
-   */
-  renderItem?: (
-    value: PickerValue,
-    itemProps: PickerItemProps & {isSelected: boolean; isItemDisabled: boolean},
-    label?: string
-  ) => React.ReactElement;
-  /**
+   * @deprecated
    * Render custom picker modal (e.g ({visible, children, toggleModal}) => {...})
```

Comment:
```suggestion
   * Render custom picker overlay (e.g ({visible, children, toggleModal}) => {...})
```

Reply:
I'll do this change in the new renamed prop - renderOverlay